File: Tutorial_Sample_Code/ReportDocObjectModel/Export/VB/VB_Win_RDObjMod_Export/Form1.vb
Imports CrystalDecisions.CrystalReports.Engine
Imports CrystalDecisions.Shared

Public Class Form1

    Private hierarchicalGroupingReport As Hierarchical_Grouping
    Private exportPath As String
    Private myDiskFileDestinationOptions As DiskFileDestinationOptions
    Private myExportOptions As ExportOptions
    Private selectedNoFormat As Boolean = True

    Private Sub ConfigureCrystalReports()
        hierarchicalGroupingReport = New Hierarchical_Grouping()
        myCrystalReportViewer.ReportSource = hierarchicalGroupingReport

        exportTypesList.DataSource = System.Enum.GetValues(GetType(ExportFormatType))
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ConfigureCrystalReports()
    End Sub

    Public Sub ExportSetup()
        exportPath = "C:\Exported\"

        If Not System.IO.Directory.Exists(exportPath) Then
            System.IO.Directory.CreateDirectory(exportPath)
        End If

        myDiskFileDestinationOptions = New DiskFileDestinationOptions()
        myExportOptions = hierarchicalGroupingReport.ExportOptions
        myExportOptions.ExportDestinationType = ExportDestinationType.DiskFile
        myExportOptions.FormatOptions = Nothing
    End Sub

    Public Sub ExportSelection()
        Select Case exportTypesList.SelectedIndex

            Case ExportFormatType.NoFormat
                selectedNoFormat = True
            Case ExportFormatType.CrystalReport
                ConfigureExportToRpt()
            Case ExportFormatType.RichText
                ConfigureExportToRtf()
            Case ExportFormatType.WordForWindows
                selectedNoFormat = True
            Case ExportFormatType.Excel
                ConfigureExportToXls()
            Case ExportFormatType.PortableDocFormat
                ConfigureExportToPdf()
            Case ExportFormatType.HTML32
                ConfigureExportToHtml32()
            Case ExportFormatType.HTML40
                ConfigureExportToHtml40()
        End Select
    End Sub

    Public Sub ExportCompletion()
        Try
            If selectedNoFormat Then
                message.Text = MessageConstants.FORMAT_NOT_SUPPORTED
            Else
                hierarchicalGroupingReport.Export()
                message.Text = MessageConstants.SUCCESS
            End If
        Catch ex As Exception
            message.Text = MessageConstants.FAILURE & ex.Message
        End Try

        message.Visible = True
        selectedNoFormat = False
    End Sub

    Public Sub ConfigureExportToRpt()
        myExportOptions.ExportFormatType = ExportFormatType.CrystalReport
        myDiskFileDestinationOptions.DiskFileName = exportPath & "Report.rpt"
        myExportOptions.DestinationOptions = myDiskFileDestinationOptions
    End Sub

    Public Sub ConfigureExportToRtf()
        myExportOptions.ExportFormatType = ExportFormatType.RichText
        myDiskFileDestinationOptions.DiskFileName = exportPath & "RichTextFormat.rtf"
        myExportOptions.DestinationOptions = myDiskFileDestinationOptions
    End Sub

    Public Sub ConfigureExportToDoc()
        myExportOptions.ExportFormatType = ExportFormatType.WordForWindows
        myDiskFileDestinationOptions.DiskFileName = exportPath & "Word.doc"
        myExportOptions.DestinationOptions = myDiskFileDestinationOptions
    End Sub

    Public Sub ConfigureExportToXls()
        myExportOptions.ExportFormatType = ExportFormatType.Excel
        myDiskFileDestinationOptions.DiskFileName = exportPath & "Excel.xls"
        myExportOptions.DestinationOptions = myDiskFileDestinationOptions
    End Sub

    Public Sub ConfigureExportToPdf()
        myExportOptions.ExportFormatType = ExportFormatType.PortableDocFormat
        myDiskFileDestinationOptions.DiskFileName = exportPath & "PortableDoc.pdf"
        myExportOptions.DestinationOptions = myDiskFileDestinationOptions
    End Sub

    Public Sub ConfigureExportToHtml32()
        myExportOptions.ExportFormatType = ExportFormatType.HTML32
        Dim html32FormatOptions As HTMLFormatOptions = New HTMLFormatOptions()
        html32FormatOptions.HTMLBaseFolderName = exportPath & "Html32Folder"
        html32FormatOptions.HTMLFileName = "html32.html"
        html32FormatOptions.HTMLEnableSeparatedPages = False
        html32FormatOptions.HTMLHasPageNavigator = False
        myExportOptions.FormatOptions = html32FormatOptions
    End Sub

    Public Sub ConfigureExportToHtml40()
        myExportOptions.ExportFormatType = ExportFormatType.HTML40
        Dim html40FormatOptions As HTMLFormatOptions = New HTMLFormatOptions()
        html40FormatOptions.HTMLBaseFolderName = exportPath & "Html40Folder"
        html40FormatOptions.HTMLFileName = "html40.html"
        html40FormatOptions.HTMLEnableSeparatedPages = True
        html40FormatOptions.HTMLHasPageNavigator = True
        html40FormatOptions.FirstPageNumber = 1
        html40FormatOptions.LastPageNumber = 3
        myExportOptions.FormatOptions = html40FormatOptions
    End Sub

    Private Sub exportByType_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles exportByType.Click
        ExportSetup()
        ExportSelection()
        ExportCompletion()
    End Sub

End Class

File: Tutorial_Sample_Code/ReportDocObjectModel/Export/CS/CS_Win_RDObjMod_Export/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Shared;

namespace ExpWinCS
{
    public partial class Form1 : Form
    {
        private Hierarchical_Grouping hierarchicalGroupingReport;
        private string exportPath;
        private DiskFileDestinationOptions diskFileDestinationOptions;
        private ExportOptions exportOptions;
        private bool selectedNoFormat = true;

        public Form1()
        {
            InitializeComponent();
        }

        private void ConfigureCrystalReports()
        {
            hierarchicalGroupingReport = new Hierarchical_Grouping();
            crystalReportViewer.ReportSource = hierarchicalGroupingReport;

            exportTypesList.DataSource = System.Enum.GetValues(typeof(ExportFormatType));
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            ConfigureCrystalReports();
        }

        private void ExportSetup()
        {
            exportPath = "C:\\Exported\\";

            if (!System.IO.Directory.Exists(exportPath))
            {
                System.IO.Directory.CreateDirectory(exportPath);
            }

            diskFileDestinationOptions = new DiskFileDestinationOptions();
            exportOptions = hierarchicalGroupingReport.ExportOptions;
            exportOptions.ExportDestinationType = ExportDestinationType.DiskFile;
            exportOptions.FormatOptions = null;

        }

        private void ExportSelection()
        {
            switch ((ExportFormatType)exportTypesList.SelectedIndex)
            {
                case ExportFormatType.NoFormat:
                    selectedNoFormat = true;
                    break;
                case ExportFormatType.CrystalReport:
                    ConfigureExportToRpt();
                    break;
                case ExportFormatType.RichText:
                    ConfigureExportToRtf();
                    break;
                case ExportFormatType.WordForWindows:
                    ConfigureExportToDoc();
                    break;
                case ExportFormatType.Excel:
                    ConfigureExportToXls();
                    break;
                case ExportFormatType.PortableDocFormat:
                    ConfigureExportToPdf();
                    break;
                case ExportFormatType.HTML32:
                    ConfigureExportToHtml32();
                    break;
                case ExportFormatType.HTML40:
                    ConfigureExportToHtml40();
                    break;
            }
        }

        private void ExportCompletion()
        {
            try
            {
                if (selectedNoFormat)
                {
                    message.Text = MessageConstants.FORMAT_NOT_SUPPORTED;
                }
                else
                {
                    hierarchicalGroupingReport.Export();
                    message.Text = MessageConstants.SUCCESS;
                }
            }
            catch (Exception ex)
            {
                message.Text = MessageConstants.FAILURE + ex.Message;
            }

            message.Visible = true;
            selectedNoFormat = false;
        }

        private void ConfigureExportToRpt()
        {
            exportOptions.ExportFormatType = ExportFormatType.CrystalReport;
            diskFileDestinationOptions.DiskFileName = exportPath + "Report.rpt";
            exportOptions.DestinationOptions = diskFileDestinationOptions;
        }

        private void ConfigureExportToRtf()
        {
            exportOptions.ExportFormatType = ExportFormatType.RichText;
            diskFileDestinationOptions.DiskFileName = exportPath + "RichTextFormat.rtf";
            exportOptions.DestinationOptions = diskFileDestinationOptions;
        }

        private void ConfigureExportToDoc()
        {
            exportOptions.ExportFormatType = ExportFormatType.WordForWindows;
            diskFileDestinationOptions.DiskFileName = exportPath + "Word.doc";
            exportOptions.DestinationOptions = diskFileDestinationOptions;
        }

        private void ConfigureExportToXls()
        {
            exportOptions.ExportFormatType = ExportFormatType.Excel;
            diskFileDestinationOptions.DiskFileName = exportPath + "Excel.xls";
            exportOptions.DestinationOptions = diskFileDestinationOptions;
        }

        private void ConfigureExportToPdf()
        {
            exportOptions.ExportFormatType = ExportFormatType.PortableDocFormat;
            diskFileDestinationOptions.DiskFileName = exportPath + "PortableDoc.pdf";
            exportOptions.DestinationOptions = diskFileDestinationOptions;
        }

        private void ConfigureExportToHtml32()
        {
            exportOptions.ExportFormatType = ExportFormatType.HTML32;
            HTMLFormatOptions html32FormatOptions = new HTMLFormatOptions();
            html32FormatOptions.HTMLBaseFolderName = exportPath + "Html32Folder";
            html32FormatOptions.HTMLFileName = "html32.html";
            html32FormatOptions.HTMLEnableSeparatedPages = false;
            html32FormatOptions.HTMLHasPageNavigator = false;
            exportOptions.FormatOptions = html32FormatOptions;
        }

        private void ConfigureExportToHtml40()
        {
            exportOptions.ExportFormatType = ExportFormatType.HTML40;
            HTMLFormatOptions html40FormatOptions = new HTMLFormatOptions();
            html40FormatOptions.HTMLBaseFolderName = exportPath + "Html40Folder";
            html40FormatOptions.HTMLFileName = "html40.html";
            html40FormatOptions.HTMLEnableSeparatedPages = true;
            html40FormatOptions.HTMLHasPageNavigator = true;
            html40FormatOptions.FirstPageNumber = 1;
            html40FormatOptions.LastPageNumber = 3;
            exportOptions.FormatOptions = html40FormatOptions;
        }

        private void exportByType_Click(object sender, EventArgs e)
        {
            ExportSetup();
            ExportSelection();
            ExportCompletion();
        }
    }
}